docs/CONFIG-VARS.md

```diff
@@ -100,6 +100,11 @@ Additional node pools can be created separate from the default nodepool. This is
 | os_disk_size | Disk size for nodepool VMs in GB | number | |
 | min_nodes | Minimum number of nodes for the nodepool | number | Value must be between 0 and 100. Setting min and max node counts the same disables autoscaling |
 | max_nodes | Maximum number of nodes for the nodepool | number | Value must be between 0 and 100. Setting min and max node counts the same disables autoscaling |
+| max_pods | Maximum number of pods per node | number | Default is 110
+| mode |  Nodepool mode | string | System or User
+| enable_node_public_ip | Enable public IP for each node | bool | |
```

Comment:
So not sure why all of these items are being surfaced now: `mode`, `enable_node_public_ip`, `priority`, and `eviction_policy`. I am thinking most of these would be nice to have, but these are not part of this pr. If we need them for this change, please add more details in the pr description. If they are not needed, then they should be removed and placed in another pr.

Reply:
does enable_node_public_ip actually make sense? The nodes sit on a private subnet, so there is no way to route to their public ip's directly anyway.

Reply:
I was thinking the same thing, but again, believe these value additions should be in another pr.

Reply:
These are to expose all available configs to the user, should they want to exercise any and all of these - https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool. I could see cloud cost conscious user's wanting to take advantage of 'Spot' VM prices.

Have to check on 'enable_node_public_ip=true' with subnet config, could be removed.

Reply:
I agree, I just don't believe it's part of this pr. They might be needed and do provide flexibility, just not answer this pr's question. My vote would be to open another pr and present these changes there.